addons/web_editor/static/src/js/wysiwyg/widgets/media.js

```diff
@@ -500,7 +500,14 @@ var FileWidget = SearchableMediaWidget.extend({
             this.media.dataset.mimetype = img.mimetype;
         }
         this.media.classList.remove('o_modified_image_to_save');
-        this.$media.trigger('image_changed');
+        // When website or mass_mailing editor is used, image_changed needs to
+        // be awaited to perform the automatic optimization of the image.
+        // To achieve this, the event is also triggered from Wysiwyg's
+        // openMediaDialog.
+        // if (!document.querySelector("#oe_snippets.o_loaded")) {
+        if (!this.options.isImageChangedTriggeredByCaller) {
```

Comment:
There is probably a better way than this... I guess to investigate in master. Meanwhile, I'll rename it into `notifyOnImageChanged`